src/components/visualizations/chart/highcharts/getAxisNameConfiguration.ts

```diff
@@ -11,19 +14,24 @@ const axisNameConfigGetter = (chartOptions: IChartOptions) => (axisNamePrefix: s
             return {};
         }
 
+        const isYAxis = axisNamePrefix === "y";
         const opposite = get(axis, "opposite", false);
-        const axisPropsKey = opposite ? `secondary_${axisNamePrefix}AxisProps` : `${axisNamePrefix}AxisProps`;
+        const axisPropsKey = opposite
+            ? `secondary_${axisNamePrefix}AxisProps.name`
+            : `${axisNamePrefix}AxisProps.name`;
 
         return {
-            title: getHighchartsAxisTitleConfiguration(chartOptions, axisPropsKey),
+            title: getHighchartsAxisTitleConfiguration(chartOptions, axisPropsKey, opposite, isYAxis),
         };
     });
 
 function getHighchartsAxisTitleConfiguration(
     chartOptions: IChartOptions,
     axisPropsKey: string,
+    opposite: boolean,
+    isYAxis: boolean,
 ): HighchartsAxisTitle {
```

Comment:
I find the signature of the method strange. I would remove `axisPropsKey`, `opposite`, and `isYAxis` parameters and instead of it provide `axis` and `axisNamePrefix` parameter. You can then resolve `opposite`, `axisPropsKey` and `isYAxis` values inside of it. It would be better to hide the logic inside of the method that needs it.